releasing.md

```diff
@@ -0,0 +1,304 @@
+digger-java-client-1.1.1-20171005.090531-4.pom.asc               Thu Oct 05 09:05:38 UTC 2017    819
+digger-java-client-1.1.1-20171005.090531-4.pom.md5               Thu Oct 05 09:05:32 UTC 2017    32
+digger-java-client-1.1.1-20171005.090531-4.pom.sha1              Thu Oct 05 09:05:32 UTC 2017    40
+```
+
+In order to use any snapshot in your dependent project, you need to have Sonatype snapshot repo in your dependent project
+as the snapshots are not synced to Maven central.
+
+```
+<repositories>
+    ...
+    <repository>
+        <id>ossrh</id>
+        <name>Sonatype OSS Snapshot Repository</name>
+        <url>http://oss.sonatype.org/content/repositories/snapshots</url>
+        <releases>
+            <enabled>false</enabled>
+        </releases>
+        <snapshots>
+            <enabled>true</enabled>
+        </snapshots>
+    </repository>
+    ...
+<repositories>
+```
+
+
+## Release process
+
+Overview:
+* We use Maven release plugin to bump versions and etc.
+* We first bump version and tag things locally, then push to GitHub, then do the release to Sonatype
+
+#### Prepare
+
+While on master branch, with a `-SNAPSHOT` appended to the version in pom.xml, run this:
+
+```
+mvn clean
+mvn release:prepare -Dtag=tag
+```
+
+_**NOTE:**_  The tag should only be the version number like `mvn release:prepare -Dtag=0.2.21`
+
+The release plugin does lots of thing for you like bumping to the next snapshot version and checking files, you can do
+a `git status` and check the differences.
+
+
+#### Sign the Git tag
+
+   * Note the commit git is pointing (commit_SHA_of_the_prepare_release): `git show --name-status x.y.z`
+   * Delete the tag : ```git tag -d x.y.z```
+   * Recreate the tag signing it : ```git tag -s x.y.z <commit_SHA_of_the_prepare_release> -m "signed MY ARTIFACT x.y.z tag"```
+   * Make sure you signed the correct commit ![commit](http://photon.abstractj.org/1._hd2opensourceaerogearagpushaerogearunifiedpushjavaclient_zsh_20130731_084730.jpg)
+
+   * Verify that you really singed the tag, by executing ```git tag -v x.y.z```. The output should be similar to:
+```
+        object ac05cd2a4ed8b6ef354acdf17fff321f33548997
+        type commit
+        tag 0.2.1
+        tagger Matthias Wessendorf <matzew@apache.org> 1375864277 +0200
+
+        signed unified push java client 0.2.1 tag
+        gpg: Signature made Wed Aug  7 10:31:17 2013 CEST using DSA key ID 1CE17EDC
+        gpg: Good signature from "Matthias Wessendorf <matzew@apache.org>"
+```
+
+   * (Optional) Go to your Github profile page and add or make sure you GPG public key is added. Going to [release tab](https://github.com/aerogear/aerogear-parent/releases/tag/0.2.21) you will show a `Verified` label.
+   * If all went OK, push the tag
+```
+git push origin x.y.z
+```
+where x.y.z is the release tag.
+
+**Note:** The delete and recreate tasks are a workaround of this [bug](http://jira.codehaus.org/browse/SCM-486) .
```

Comment:
is that still a thing ?

Reply:
I think so. The tag is not signed w/o that workaround.
That JIRA link is dead... So, cannot check if that's fixed in a future version.
Let's keep it as is: I don't want to do modifications in parent pom(s)

Reply:
Yeap, it still needed!